### PR TITLE
Prepare Docker image for `reprepro`

### PR DESCRIPTION
**Pull request by noisersup**
Closes FerretDB/Engineering#186.

### CODE REVIEW

File: ferretdb-repo/reprepro/docker-entrypoint.sh
Comment (AlekSi) [inline ferretdb-repo/reprepro/docker-entrypoint.sh:1]: Note that all other tools in that repo have only the tool itself in the ENTRYPOINT, with the `-h` / `--help` flag in CMD. There is not need to write bash.

**Review by AlekSi — COMMENTED** [review]
From the issue description:

> See existing directories with Dockerfile and Makefile for examples.

There is no Makefile at all, and the entrypoint is very different.

File: ferretdb-repo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+RUN mkdir -p /var/inputs
+RUN mkdir -p $REPREPRO_BASE_DIR
+
+COPY apt $REPREPRO_BASE_DIR/conf
+
+RUN apt-get update && \
+  apt-get install -yq ssh && \
+  apt-get -t experimental install -yq reprepro
+
+EXPOSE 22
```
Comment (chilagrow) [inline ferretdb-repo/Dockerfile:14]: do we use ssh for docker to host? a comment would be useful

**Review by chilagrow — COMMENTED** [review]
I lack full context to give complete review.

File: ferretdb-reprepro/Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+build-init:
+	docker buildx create --driver=docker-container --name=ferretdb-deps
+
+build-local:
+	docker buildx build --builder=ferretdb-deps \
+		--tag=ferretdb-reprepro-local \
+		--load .
+
+build-push:
+	rm -fr node_modules
```
Comment (noisersup) [inline ferretdb-reprepro/Makefile:10]: Are we going to use npm anywhere here?

File: ferretdb-repo/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM debian:experimental-20240722
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply ferretdb-repo/Dockerfile]: There is some functionality (support for multiple signatures, I think?) that is not available in the stable Debian release yet

File: ferretdb-repo/apt/distributions [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Origin: FerretDB
+Label: FerretDB
+Codename: bookworm
+Architectures: armhf arm64 amd64
+Components: main
+Description: Apt repository for FerretDB
+Limit: 0
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply ferretdb-repo/apt/distributions]: This file should not be there

File: ferretdb-repo/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+ENV REPREPRO_BASE_DIR=/srv/repos/apt/debian
+
+RUN mkdir -p /var/inputs
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply ferretdb-repo/Dockerfile]: It should not be there

File: ferretdb-repo/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+COPY apt $REPREPRO_BASE_DIR/conf
+
+RUN apt-get update && \
+  apt-get install -yq ssh && \
+  apt-get -t experimental install -yq reprepro
+
+EXPOSE 22
+ENTRYPOINT ["/bin/reprepro"]
+CMD ["-h"]
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply ferretdb-repo/Dockerfile]: `/bin/reprepro -h` shows help

File: ferretdb-repo/Dockerfile (hunk 1 above)
Comment (AlekSi) [reply ferretdb-repo/Dockerfile]: For passing GPG socket through SSH

**Pull request merged**